src-java/kilda-model/src/main/java/org/openkilda/model/SwitchConnect.java

```diff
@@ -174,6 +187,16 @@ public SwitchId getOwnerSwitchId() {
          */
         public SwitchConnectData deepCopy(SwitchConnectData source) {
             SwitchConnectDataImpl result = new SwitchConnectDataImpl();
+            String field = null;
+            try {
+                field = "switchAddress";
+                result.setSwitchAddress(source.getSwitchAddress());
+                field = "speakerAddress";
+                result.setSpeakerAddress(source.getSpeakerAddress());
+            } catch (InvalidDataException e) {
+                throw new IllegalStateException(String.format(
```

Comment:
1. What is the reason to drop the original exception? Why can't it be wrapped as the cause?
2.  Why can't setSwitchAddress and setSpeakerAddress methods indicate the field name which causes the exception - this will simplify the code by removing "field". As for now, setting and handling "field" variable blurs the main logic with exception handling.

Reply:
Briliant ideas. Updating.